### PR TITLE
NoneType/Empty Elements

### LINKED ISSUE #22: Add more test covering empty/None attributes
I noticed some irregularities when using the converter on files, which do not have entries for all attributes. E.g. the following test fails:

```python
    def test_empty(self):
        odml.Property(name='empty property', value=[],
                      parent=self.odml_doc.sections[0],
                      unit=None, uncertainty=None, reference=None)

        self.test_double_conversion()
```

### PR DESCRIPTION
**Pull request by fschrader1992**
This PR:

- adds tests for None/empty elements/attributes. Closes #22 
- fixes some minor warnings and style issues

### CODE REVIEW

**Comment by fschrader1992** [comment]
Checks for Code Quality:
- [x] Pylint
- [x] Rebase onto GNODE Master
- [x] Repeat Local Tests Python 3.8

File: nixodmlconverter/convert.py [hunk 1]
```diff
@@ -197,12 +197,15 @@ def odml_to_nix_property(odmlprop, nixsec):
...
+    #if not nixvalues:
+    #    INFO["skipped empty properties"] += 1
+    #    return
 
     # We need to get the appropriate NIX DataType for the current odML values
-    dtype = nix.DataType.get_dtype(nixvalues[0])
+    if nixvalues:
+        dtype = nix.DataType.get_dtype(nixvalues[0])
+    else:
+        dtype = nix.DataType.get_dtype("")
```
Comment (achilleas-k) [inline nixodmlconverter/convert.py]: Would it be simpler to set the dtype directly to `nix.DataType.String` and avoid needing to run the detection function?  It's not costly or anything, but perhaps more readable and explicit.

**Review by achilleas-k — APPROVED** [review]
One minor comment.

**Comment by fschrader1992** [comment]
Checks for Code Quality:
- [x] Pylint
- [x] Rebase onto GNODE Master
- [x] Repeat Local Tests Python 3.8

**Review by achilleas-k — APPROVED** [review]

**Review by mpsonntag — APPROVED** [review]
LGTM
